New Package - Phase2 L1CaloTrigger

rekovic opened the pull request:
New Package for L1T Phase2: Only Standalone EG in Barrel and EndCap, 


#### PR description:

 - addressed comments,  with data files removed.
 - depends on externals cms-data/L1Trigger-L1TCalorimeter#22
 - supersedes https://github.com/cms-sw/cmssw/pull/29904

rekovic committed:
Phase2 L1CaloTrigger - only Standalone EG in Barrel and EndCap, addressed comments,  with data files removed.

rekovic commented:
assign upgrade

cmsbuild commented:
The code-checks are being triggered in jenkins.

rekovic commented:
assign l1

rekovic commented:
test parameters

 * pull_request = cms-data/L1Trigger-L1TCalorimeter#22

rekovic commented:
please test

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-30325/16317

- This PR adds an extra 36KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  L1Trigger/Configuration/python/SimL1Emulator_cff.py  modified in PR(s): #29904, #30323
  - File  L1Trigger/L1CaloTrigger/BuildFile.xml  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/interface/L1EGammaEECalibrator.h  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/interface/ParametricCalibration.h  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/plugins/BuildFile.xml  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/plugins/L1EGammaCrystalsEmulatorProducer.cc  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/plugins/L1EGammaEEProducer.cc  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/python/L1EGammaCrystalsEmulatorProducer_cfi.py  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/python/L1TowerCalibrationProducer_cfi.py  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/python/l1EGammaCrystalsProducer_cfi.py  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/python/l1EGammaEEProducer_cfi.py  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/python/ntuple_cfi.py  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/python/test_L1EGCrystalClusterEmulator_cfg.py  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/src/L1EGammaEECalibrator.cc  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/src/ParametricCalibration.cc  modified in PR(s): #29904

cmsbuild commented:
New categories assigned: upgrade

@kpedro88 you have been requested to review this Pull request/Issue and eventually sign? Thanks

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
A new Pull Request was created by  @rekovic  for master.

It involves the following packages:

L1Trigger/Configuration
L1Trigger/L1CaloTrigger


The following packages do not have a category, yet:

L1Trigger/L1CaloTrigger
Please create a PR for https://github.com/cms-sw/cms-bot/blob/master/categories_map.py to assign category

@cmsbuild, @rekovic, @benkrikler, @kpedro88 can you please review it and eventually sign? Thanks.
@Martin-Grunewald this is something you requested to watch as well.
@silviodonato, @dpiparo you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

rekovic commented on L1Trigger/L1CaloTrigger/plugins/L1EGammaEEProducer.cc:
```diff
@@ -0,0 +1,188 @@
+
+  // here we loop on the TPGs
+  for (auto cl3d = multiclusters.begin(0); cl3d != multiclusters.end(0); cl3d++) {
+    // std::cout << "-- CL3D is EG: " <<  cl3d->hwQual() << "   "<< cl3d->eta() <<"   "<< cl3d->pt()<<std::endl;
```
lets leave this commented code as is, useful for checks and validation.
Will clean up in future PRs.

kpedro88 replied:
if there is a reason to keep commented-out code, that reason should be indicated clearly (in a separate comment). In many cases, I think the commented-out code in this PR is not needed. In general, things that are not cleaned up in the initial PR tend to persist forever.

cmsbuild commented:
-1

Tested at: c67896b5c0fdecc5160eb6f29e39bb746f15f36a

CMSSW: CMSSW_11_2_X_2020-06-21-2300
SCRAM_ARCH: slc7_amd64_gcc820
You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cd687b/7240/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

When I ran the RelVals I found an error in the following workflows:
20034.0 step2
<pre>runTheMatrix-results/20034.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D35_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D35+RecoFullGlobal_2026D35+HARVESTFullGlobal_2026D35/step2_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D35_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D35+RecoFullGlobal_2026D35+HARVESTFullGlobal_2026D35.log

</pre>20434.0 step2
<pre>runTheMatrix-results/20434.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D41_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D41+RecoFullGlobal_2026D41+HARVESTFullGlobal_2026D41/step2_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D41_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D41+RecoFullGlobal_2026D41+HARVESTFullGlobal_2026D41.log

</pre>23234.0 step2
<pre>runTheMatrix-results/23234.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D49_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D49+RecoFullGlobal_2026D49+HARVESTFullGlobal_2026D49/step2_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D49_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D49+RecoFullGlobal_2026D49+HARVESTFullGlobal_2026D49.log

</pre>21234.0 step2
<pre>runTheMatrix-results/21234.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D44_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D44+RecoFullGlobal_2026D44+HARVESTFullGlobal_2026D44/step2_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D44_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D44+RecoFullGlobal_2026D44+HARVESTFullGlobal_2026D44.log

</pre>

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

rekovic commented on L1Trigger/L1CaloTrigger/plugins/L1EGammaEEProducer.cc:
```diff
@@ -0,0 +1,188 @@
+    }
+  }
+
+  // std::cout << "# of selected HGCAL multiclusters: " << selected_multiclusters.size() << std::endl;
```
lets leave this commented code as is, useful for checks and validation.
Will clean up in future PRs.

rekovic commented on L1Trigger/L1CaloTrigger/plugins/L1EGammaEEProducer.cc:
```diff
@@ -0,0 +1,188 @@
+      // this is not yet used
+      used_clusters.insert(cl3d);
+      auto eta_phi_bin = get_eta_phi_bin(cl3d);
+      // std::cout << " cl pt: " << cl3d->pt()
```
lets leave this commented code as is, useful for checks and validation.
Will clean up in future PRs.

rekovic replied:
done

rekovic commented on L1Trigger/L1CaloTrigger/plugins/L1EGammaEEProducer.cc:
```diff
@@ -0,0 +1,188 @@
+              if (used_clusters.find(other_cl_ptr) == used_clusters.end()) {
+                if (std::abs(other_cl_ptr->eta() - cl3d->eta()) < 0.02) {
+                  if (std::abs(reco::deltaPhi(other_cl_ptr->phi(), cl3d->phi())) < 0.1) {
+                    // std::cout << "MERGE with cl pt: " << other_cl_ptr->pt()
```
lets leave this commented code as is, useful for checks and validation.
Will clean up in future PRs.

rekovic replied:
done

rekovic commented on L1Trigger/L1CaloTrigger/src/L1EGammaEECalibrator.cc:
```diff
@@ -0,0 +1,64 @@
+  if (bin_eta == -1 || bin_pt == -1)
+    return 1.;
+  auto n_bins_pt = pt_bins.size();
+  // std::cout << "pt: " << pt << " eta: " << eta << " ptbin: " << bin_pt << " etabin: " << bin_eta << " calib: " <<  calib_factors[(bin_eta*n_bins_pt)+bin_pt] << std::endl;
```
lets leave this commented code as is, useful for checks and validation.
Will clean up in future PRs.

rekovic commented:
@smuzaffar 
Strange.  I copied your syntax from https://github.com/cms-sw/cmssw/pull/29904#issuecomment-647282374 which worked and the test jobs were successful to here https://github.com/cms-sw/cmssw/pull/30325#issuecomment-647398279
 but here it failed as the path to the data file could not be found.  

Was there some subtlety in the syntax that I missed again ?

smuzaffar commented:
@rekovic , you used '*' instead of '-' :-)

smuzaffar commented:
I will update the bot to accept both but for now please use `-`

smuzaffar commented:
please test

bot should now recognize `*` in test parameters

cmsbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) cms-data/L1Trigger-L1TCalorimeter#22

cmsbuild commented:
+1
Tested at: c67896b5c0fdecc5160eb6f29e39bb746f15f36a
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c806ed/7242/summary.html
CMSSW: CMSSW_11_2_X_2020-06-21-2300
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c806ed/7242/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 4 differences found in the comparisons
 - DQMHistoTests: Total files compared: 36
 - DQMHistoTests: Total histograms compared: 2778811
 - DQMHistoTests: Total failures: 4
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2778757
 - DQMHistoTests: Total skipped: 50
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 35 files compared)
 - Checked 152 log files, 16 edm output root files, 36 DQM output files

rekovic commented:
+1

silviodonato commented:
urgent

kpedro88 commented:
-1
many comments from #29904 not addressed at all
i'm not going through and repeating all the comments

rekovic commented:
@kpedro88 
I believe that a huge portion of of them are addressed, and maybe a handful are not. Will go back to #29904 and explicitly mark all the addressed comments.  We can continue the discussion there on the non-addressed comments and address them if needed.

rekovic commented:
@kpedro88 
My bad.  Yes, indeed there are still many remaining comments.  I thought I had most of them, must have not seen them under the folded part.   Addressing now.

rekovic committed:
Addressed comments of #29904 (b4fdcc3)

rekovic commented:
please test

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-30325/16410

- This PR adds an extra 16KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  L1Trigger/Configuration/python/SimL1Emulator_cff.py  modified in PR(s): #29904, #30323
  - File  L1Trigger/L1CaloTrigger/BuildFile.xml  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/interface/L1EGammaEECalibrator.h  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/interface/ParametricCalibration.h  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/plugins/BuildFile.xml  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/plugins/L1EGammaCrystalsEmulatorProducer.cc  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/plugins/L1EGammaEEProducer.cc  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/python/L1EGammaCrystalsEmulatorProducer_cfi.py  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/python/L1TowerCalibrationProducer_cfi.py  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/python/l1EGammaCrystalsProducer_cfi.py  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/python/l1EGammaEEProducer_cfi.py  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/python/ntuple_cfi.py  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/python/test_L1EGCrystalClusterEmulator_cfg.py  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/src/L1EGammaEECalibrator.cc  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/src/ParametricCalibration.cc  modified in PR(s): #29904

cmsbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) cms-data/L1Trigger-L1TCalorimeter#22

cmsbuild commented:
Pull request #30325 was updated. @cmsbuild, @rekovic, @benkrikler, @kpedro88 can you please check and sign again.

cmsbuild commented:
-1

Tested at: 32952c6778b214f6be723ed3c7b2d49612cca57e

CMSSW: CMSSW_11_2_X_2020-06-24-2300
SCRAM_ARCH: slc7_amd64_gcc820
You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c806ed/7333/summary.html

I found follow errors while testing this PR

**Failed tests:  ClangBuild**

* **Clang**:

I found compilation warning while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 32 COMPILER='llvm compile'

```
 See details on the summary page.

cmsbuild commented:
Comparison not run due to Build errors/Fireworks only changes/No short matrix requested (RelVals and Igprof tests were also skipped)

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-30325/16414

- This PR adds an extra 44KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  L1Trigger/Configuration/python/SimL1Emulator_cff.py  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/BuildFile.xml  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/interface/L1EGammaEECalibrator.h  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/interface/ParametricCalibration.h  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/plugins/BuildFile.xml  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/plugins/L1EGammaCrystalsEmulatorProducer.cc  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/plugins/L1EGammaEEProducer.cc  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/python/L1EGammaCrystalsEmulatorProducer_cfi.py  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/python/L1TowerCalibrationProducer_cfi.py  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/python/l1EGammaCrystalsProducer_cfi.py  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/python/l1EGammaEEProducer_cfi.py  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/python/ntuple_cfi.py  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/python/test_L1EGCrystalClusterEmulator_cfg.py  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/src/L1EGammaEECalibrator.cc  modified in PR(s): #29904
  - File  L1Trigger/L1CaloTrigger/src/ParametricCalibration.cc  modified in PR(s): #29904

cmsbuild commented:
Pull request #30325 was updated. @cmsbuild, @rekovic, @benkrikler, @kpedro88 can you please check and sign again.

cmsbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) cms-data/L1Trigger-L1TCalorimeter#22

cmsbuild commented:
-1

Tested at: f8e11f710e159b32fc0f622ea2a96069716b2841

CMSSW: CMSSW_11_2_X_2020-06-24-2300
SCRAM_ARCH: slc7_amd64_gcc820

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - #30323 @bsunanda: Run4-hgx247 Add some new features for HGCal

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c806ed/7337/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c806ed/7337/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c806ed/7337/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

When I ran the RelVals I found an error in the following workflows:
20034.0 step2
<pre>runTheMatrix-results/20034.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D35_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D35+RecoFullGlobal_2026D35+HARVESTFullGlobal_2026D35/step2_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D35_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D35+RecoFullGlobal_2026D35+HARVESTFullGlobal_2026D35.log

</pre>20434.0 step2
<pre>runTheMatrix-results/20434.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D41_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D41+RecoFullGlobal_2026D41+HARVESTFullGlobal_2026D41/step2_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D41_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D41+RecoFullGlobal_2026D41+HARVESTFullGlobal_2026D41.log

</pre>23234.0 step2
<pre>runTheMatrix-results/23234.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D49_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D49+RecoFullGlobal_2026D49+HARVESTFullGlobal_2026D49/step2_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D49_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D49+RecoFullGlobal_2026D49+HARVESTFullGlobal_2026D49.log

</pre>21234.0 step2
<pre>runTheMatrix-results/21234.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D44_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D44+RecoFullGlobal_2026D44+HARVESTFullGlobal_2026D44/step2_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D44_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D44+RecoFullGlobal_2026D44+HARVESTFullGlobal_2026D44.log

</pre>


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - #30323 @bsunanda: Run4-hgx247 Add some new features for HGCal

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c806ed/7337/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c806ed/7337/git-merge-result

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

cmsbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) cms-sw/cmssw#30363,cms-data/L1Trigger-L1TCalorimeter#22,cms-data/L1Trigger-L1TMuon#18,cms-sw/cmssw#30362

rekovic committed:
Remove nonneeded config params.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
-1
@rekovic This pull request cannot be automatically  merged, could you please rebase it?

You can see the log for git cms-merge-topic here: 
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-30325/7341/git-merge-result

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-30325/16427

- This PR adds an extra 44KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  L1Trigger/Configuration/python/SimL1Emulator_cff.py  modified in PR(s): #30363

cmsbuild commented:
Pull request #30325 was updated. @cmsbuild, @rekovic, @benkrikler, @kpedro88 can you please check and sign again.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-30325/16428

- This PR adds an extra 44KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  L1Trigger/Configuration/python/SimL1Emulator_cff.py  modified in PR(s): #30363

cmsbuild commented:
Pull request #30325 was updated. @cmsbuild, @rekovic, @benkrikler, @kpedro88 can you please check and sign again.

rekovic commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) cms-data/L1Trigger-L1TCalorimeter#22

cmsbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) cms-sw/cmssw#30363,cms-data/L1Trigger-L1TCalorimeter#22,cms-data/L1Trigger-L1TMuon#18,cms-sw/cmssw#30362

kpedro88 commented on L1Trigger/L1CaloTrigger/plugins/L1EGammaEEProducer.cc:
```diff
@@ -0,0 +1,169 @@
+
+  std::sort(selected_multiclusters.begin(), selected_multiclusters.end(), l1tp2::compare_cluster_pt);
+  std::set<const l1t::HGCalMulticluster *> used_clusters;
+  for (const auto cl3d : selected_multiclusters) {
```
`const auto&`
(`const auto` makes a copy, which is slow and unnecessary)

rekovic replied:
done

kpedro88 commented on L1Trigger/L1CaloTrigger/plugins/L1EGammaEEProducer.cc:
```diff
@@ -0,0 +1,169 @@
+          auto bucket = etaphi_bins.find(std::make_pair(eta_bin, phi_bin));
+          if (bucket != etaphi_bins.end()) {
+            // this bucket is not empty
+            for (const auto other_cl_ptr : bucket->second) {
```
`const auto&`

rekovic replied:
done.

kpedro88 commented on L1Trigger/L1CaloTrigger/plugins/L1EGammaCrystalsEmulatorProducer.cc:
```diff
@@ -0,0 +1,1199 @@
+
+        // Add L1EGs if they match in iEta / iPhi
+        // L1EGs are already pT ordered, we will take the ID info for the leading one, but pT as the sum
+        for (auto l1eg : *L1EGXtalClusters) {
```
this is not the correct fix for the clang warning
it is telling you to use `const auto&` (reference, not copy)

rekovic replied:
done

rekovic committed:
const auto&.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
-1
@rekovic This pull request cannot be automatically  merged, could you please rebase it?

You can see the log for git cms-merge-topic here: 
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-30325/7345/git-merge-result

rekovic commented:
hmm... not sure why it cannot automatically merge.
The last thing done was force push of the rebased version.

@kpedro88  any idea ?

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-30325/16482

- This PR adds an extra 28KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  L1Trigger/Configuration/python/SimL1Emulator_cff.py  modified in PR(s): #30363


Code check has found code style and quality issues which could be resolved by applying following patch(s)
- **code-format**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-30325/16482/code-format.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-30325/16482/code-format.patch | patch -p1`
You can also run `scram build code-format` to apply code format directly

rekovic commented:
the problem is with the first commit.

cmsbuild commented:
The code-checks are being triggered in jenkins.

rekovic committed:
code format.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-30325/16488

- This PR adds an extra 40KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  L1Trigger/Configuration/python/SimL1Emulator_cff.py  modified in PR(s): #30363

cmsbuild commented:
Pull request #30325 was updated. @cmsbuild, @rekovic, @benkrikler, @kpedro88 can you please check and sign again.

rekovic commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) cms-data/L1Trigger-L1TCalorimeter#22

cmsbuild commented:
-1

Tested at: 01f1e7da3c3d0ea74cc957567dff4e706795b760

CMSSW: CMSSW_11_2_X_2020-06-25-1100
SCRAM_ARCH: slc7_amd64_gcc820
You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c806ed/7350/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals  AddOn**

* **RelVals**:

When I ran the RelVals I found an error in the following workflows:
5.1 step1
<pre>runTheMatrix-results/5.1_TTbar+TTbarFS+HARVESTFS/step1_TTbar+TTbarFS+HARVESTFS.log

</pre>135.4 step1
<pre>runTheMatrix-results/135.4_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS/step1_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS.log

</pre>8.0 step2
<pre>runTheMatrix-results/8.0_BeamHalo+BeamHalo+DIGICOS+RECOCOS+ALCABH+HARVESTCOS/step2_BeamHalo+BeamHalo+DIGICOS+RECOCOS+ALCABH+HARVESTCOS.log

</pre>7.3 step2
<pre>runTheMatrix-results/7.3_CosmicsSPLoose_UP18+CosmicsSPLoose_UP18+DIGICOS_UP18+RECOCOS_UP18+ALCACOS_UP18+HARVESTCOS_UP18/step2_CosmicsSPLoose_UP18+CosmicsSPLoose_UP18+DIGICOS_UP18+RECOCOS_UP18+ALCACOS_UP18+HARVESTCOS_UP18.log

</pre>1306.0 step2
<pre>runTheMatrix-results/1306.0_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15/step2_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>9.0 step2
<pre>runTheMatrix-results/9.0_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST/step2_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST.log

</pre>25.0 step2
<pre>runTheMatrix-results/25.0_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT/step2_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT.log

</pre>158.0 step2
<pre>runTheMatrix-results/158.0_HydjetQ_B12_5020GeV_2018_ppReco+HydjetQ_B12_5020GeV_2018_ppReco+DIGIHI2018PPRECO+RECOHI2018PPRECO+ALCARECOHI2018PPRECO+HARVESTHI2018PPRECO/step2_HydjetQ_B12_5020GeV_2018_ppReco+HydjetQ_B12_5020GeV_2018_ppReco+DIGIHI2018PPRECO+RECOHI2018PPRECO+ALCARECOHI2018PPRECO+HARVESTHI2018PPRECO.log

</pre>1330.0 step2
<pre>runTheMatrix-results/1330.0_ZMM_13+ZMM_13+DIGIUP15+RECOUP15_L1TMuDQM+HARVESTUP15_L1TMuDQM+NANOUP15/step2_ZMM_13+ZMM_13+DIGIUP15+RECOUP15_L1TMuDQM+HARVESTUP15_L1TMuDQM+NANOUP15.log

</pre>10042.0 step2
<pre>runTheMatrix-results/10042.0_ZMM_13+ZMM_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017+ALCAFull_2017+NanoFull_2017/step2_ZMM_13+ZMM_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017+ALCAFull_2017+NanoFull_2017.log

</pre>25202.0 step2
<pre>runTheMatrix-results/25202.0_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25+NANOUP15_PU25/step2_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25+NANOUP15_PU25.log

</pre>10024.0 step2
<pre>runTheMatrix-results/10024.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017+ALCAFull_2017+NanoFull_2017/step2_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017+ALCAFull_2017+NanoFull_2017.log

</pre>10224.0 step2
<pre>runTheMatrix-results/10224.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017PU_GenSimFull+DigiFullPU_2017PU+RecoFullPU_2017PU+HARVESTFullPU_2017PU+NanoFull_2017PU/step2_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017PU_GenSimFull+DigiFullPU_2017PU+RecoFullPU_2017PU+HARVESTFullPU_2017PU+NanoFull_2017PU.log

</pre>10824.0 step2
<pre>runTheMatrix-results/10824.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_2018+HARVESTFull_2018+ALCAFull_2018+NanoFull_2018/step2_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_2018+HARVESTFull_2018+ALCAFull_2018+NanoFull_2018.log

</pre>11634.0 step2
<pre>runTheMatrix-results/11634.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2021_GenSimFull+DigiFull_2021+RecoFull_2021+HARVESTFull_2021+ALCAFull_2021/step2_TTbar_14TeV+TTbar_14TeV_TuneCP5_2021_GenSimFull+DigiFull_2021+RecoFull_2021+HARVESTFull_2021+ALCAFull_2021.log

</pre>12434.0 step2
<pre>runTheMatrix-results/12434.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2023_GenSimFull+DigiFull_2023+RecoFull_2023+HARVESTFull_2023+ALCAFull_2023/step2_TTbar_14TeV+TTbar_14TeV_TuneCP5_2023_GenSimFull+DigiFull_2023+RecoFull_2023+HARVESTFull_2023+ALCAFull_2023.log

</pre>20034.0 step2
<pre>runTheMatrix-results/20034.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D35_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D35+RecoFullGlobal_2026D35+HARVESTFullGlobal_2026D35/step2_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D35_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D35+RecoFullGlobal_2026D35+HARVESTFullGlobal_2026D35.log

</pre>20434.0 step2
<pre>runTheMatrix-results/20434.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D41_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D41+RecoFullGlobal_2026D41+HARVESTFullGlobal_2026D41/step2_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D41_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D41+RecoFullGlobal_2026D41+HARVESTFullGlobal_2026D41.log

</pre>23234.0 step2
<pre>runTheMatrix-results/23234.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D49_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D49+RecoFullGlobal_2026D49+HARVESTFullGlobal_2026D49/step2_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D49_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D49+RecoFullGlobal_2026D49+HARVESTFullGlobal_2026D49.log

</pre>21234.0 step2
<pre>runTheMatrix-results/21234.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D44_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D44+RecoFullGlobal_2026D44+HARVESTFullGlobal_2026D44/step2_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D44_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D44+RecoFullGlobal_2026D44+HARVESTFullGlobal_2026D44.log

</pre>250202.181 step3
<pre>runTheMatrix-results/250202.181_TTbar_13UP18+TTbar_13UP18+PREMIXUP18_PU25+DIGIPRMXLOCALUP18_PU25+RECOPRMXUP18_PU25+HARVESTUP18_PU25/step3_TTbar_13UP18+TTbar_13UP18+PREMIXUP18_PU25+DIGIPRMXLOCALUP18_PU25+RECOPRMXUP18_PU25+HARVESTUP18_PU25.log

</pre>

* **AddOn**:

I found errors in the following addon tests:

cmsDriver.py TTbar_8TeV_TuneCUETP8M1_cfi  --conditions auto:run1_mc --fast  -n 100 --eventcontent AODSIM,DQM --relval 100000,1000 -s GEN,SIM,RECOBEFMIX,DIGI:pdigi_valid,L1,DIGI2RAW,L1Reco,RECO,EI,VALIDATION  --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --datatier GEN-SIM-DIGI-RECO,DQMIO --beamspot Realistic8TeVCollision : FAILED - time: date Thu Jun 25 18:08:04 2020-date Thu Jun 25 18:07:59 2020 s - exit: 256
cmsDriver.py TTbar_Tauola_13TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW --mc --scenario=pp -n 10 --conditions auto:run2_mc_Fake1 --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_25ns --fileout file:RelVal_Raw_Fake1_MC.root : FAILED - time: date Thu Jun 25 18:08:39 2020-date Thu Jun 25 18:08:10 2020 s - exit: 256
cmsRun /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc820/cms/cmssw-patch/CMSSW_11_2_X_2020-06-25-1100/src/HLTrigger/Configuration/test/OnLine_HLT_Fake1.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Thu Jun 25 18:08:39 2020-date Thu Jun 25 18:08:10 2020 s - exit: 21504
cmsDriver.py RelVal -s HLT:Fake1,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_Fake1 --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_25ns --processName=HLTRECO --filein file:RelVal_Raw_Fake1_MC.root --fileout file:RelVal_Raw_Fake1_MC_HLT_RECO.root : FAILED - time: date Thu Jun 25 18:08:39 2020-date Thu Jun 25 18:08:10 2020 s - exit: 21504
cmsDriver.py TTbar_Tauola_13TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW --mc --scenario=pp -n 10 --conditions auto:run2_mc_2018 --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2018 --fileout file:RelVal_Raw_2018_MC.root : FAILED - time: date Thu Jun 25 18:09:41 2020-date Thu Jun 25 18:08:15 2020 s - exit: 256
cmsRun /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc820/cms/cmssw-patch/CMSSW_11_2_X_2020-06-25-1100/src/HLTrigger/Configuration/test/OnLine_HLT_2018.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Thu Jun 25 18:09:41 2020-date Thu Jun 25 18:08:15 2020 s - exit: 21504
cmsDriver.py RelVal -s HLT:2018,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_2018 --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2018 --processName=HLTRECO --filein file:RelVal_Raw_2018_MC.root --fileout file:RelVal_Raw_2018_MC_HLT_RECO.root : FAILED - time: date Thu Jun 25 18:09:41 2020-date Thu Jun 25 18:08:15 2020 s - exit: 21504
cmsDriver.py TTbar_Tauola_13TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW --mc --scenario=pp -n 10 --conditions auto:run3_mc_GRun --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run3 --fileout file:RelVal_Raw_GRun_MC.root : FAILED - time: date Thu Jun 25 18:09:50 2020-date Thu Jun 25 18:08:24 2020 s - exit: 256
cmsRun /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc820/cms/cmssw-patch/CMSSW_11_2_X_2020-06-25-1100/src/HLTrigger/Configuration/test/OnLine_HLT_GRun.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Thu Jun 25 18:09:50 2020-date Thu Jun 25 18:08:24 2020 s - exit: 21504
cmsDriver.py RelVal -s HLT:GRun,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run3_mc_GRun --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run3 --processName=HLTRECO --filein file:RelVal_Raw_GRun_MC.root --fileout file:RelVal_Raw_GRun_MC_HLT_RECO.root : FAILED - time: date Thu Jun 25 18:09:50 2020-date Thu Jun 25 18:08:24 2020 s - exit: 21504
cmsDriver.py TTbar_13TeV_TuneCUETP8M1_cfi --conditions auto:run2_mc_l1stage1 --fast  -n 100 --eventcontent AODSIM,DQM --relval 100000,1000 -s GEN,SIM,RECOBEFMIX,DIGI:pdigi_valid,L1,DIGI2RAW,L1Reco,RECO,EI,VALIDATION  --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --datatier GEN-SIM-DIGI-RECO,DQMIO --beamspot NominalCollision2015 --era Run2_25ns : FAILED - time: date Thu Jun 25 18:08:33 2020-date Thu Jun 25 18:08:29 2020 s - exit: 256
cmsDriver.py TTbar_Tauola_13TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW --mc --scenario=pp -n 10 --conditions auto:run3_mc_PRef --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run3 --fileout file:RelVal_Raw_PRef_MC.root : FAILED - time: date Thu Jun 25 18:09:18 2020-date Thu Jun 25 18:08:34 2020 s - exit: 256
cmsRun /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc820/cms/cmssw-patch/CMSSW_11_2_X_2020-06-25-1100/src/HLTrigger/Configuration/test/OnLine_HLT_PRef.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Thu Jun 25 18:09:18 2020-date Thu Jun 25 18:08:34 2020 s - exit: 21504
cmsDriver.py RelVal -s HLT:PRef,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run3_mc_PRef --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run3 --processName=HLTRECO --filein file:RelVal_Raw_PRef_MC.root --fileout file:RelVal_Raw_PRef_MC_HLT_RECO.root : FAILED - time: date Thu Jun 25 18:09:18 2020-date Thu Jun 25 18:08:34 2020 s - exit: 21504
cmsDriver.py TTbar_13TeV_TuneCUETP8M1_cfi --conditions auto:run2_mc --fast  -n 100 --eventcontent AODSIM,DQM --relval 100000,1000 -s GEN,SIM,RECOBEFMIX,DIGI:pdigi_valid,L1,DIGI2RAW,L1Reco,RECO,EI,VALIDATION  --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --datatier GEN-SIM-DIGI-RECO,DQMIO --beamspot NominalCollision2015 --era Run2_2016 : FAILED - time: date Thu Jun 25 18:08:39 2020-date Thu Jun 25 18:08:35 2020 s - exit: 256
cmsDriver.py TTbar_Tauola_8TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW  --mc --scenario=pp -n 10 --conditions auto:run1_mc_Fake --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --fileout file:RelVal_Raw_Fake_MC.root : FAILED - time: date Thu Jun 25 18:09:19 2020-date Thu Jun 25 18:08:40 2020 s - exit: 256
cmsRun /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc820/cms/cmssw-patch/CMSSW_11_2_X_2020-06-25-1100/src/HLTrigger/Configuration/test/OnLine_HLT_Fake.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Thu Jun 25 18:09:19 2020-date Thu Jun 25 18:08:40 2020 s - exit: 21504
cmsDriver.py RelVal -s HLT:Fake,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run1_mc_Fake --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --processName=HLTRECO --filein file:RelVal_Raw_Fake_MC.root --fileout file:RelVal_Raw_Fake_MC_HLT_RECO.root : FAILED - time: date Thu Jun 25 18:09:19 2020-date Thu Jun 25 18:08:40 2020 s - exit: 21504
cmsDriver.py TTbar_Tauola_13TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW --mc --scenario=pp -n 10 --conditions auto:run3_mc_HIon --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run3_pp_on_PbPb --fileout file:RelVal_Raw_HIon_MC.root : FAILED - time: date Thu Jun 25 18:09:50 2020-date Thu Jun 25 18:08:45 2020 s - exit: 256
cmsRun /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc820/cms/cmssw-patch/CMSSW_11_2_X_2020-06-25-1100/src/HLTrigger/Configuration/test/OnLine_HLT_HIon.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Thu Jun 25 18:09:50 2020-date Thu Jun 25 18:08:45 2020 s - exit: 21504
cmsDriver.py RelVal -s HLT:HIon,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run3_mc_HIon --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run3_pp_on_PbPb --processName=HLTRECO --filein file:RelVal_Raw_HIon_MC.root --fileout file:RelVal_Raw_HIon_MC_HLT_RECO.root : FAILED - time: date Thu Jun 25 18:09:50 2020-date Thu Jun 25 18:08:45 2020 s - exit: 21504
cmsDriver.py TTbar_Tauola_13TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW --mc --scenario=pp -n 10 --conditions auto:run2_mc_Fake2 --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2016 --fileout file:RelVal_Raw_Fake2_MC.root : FAILED - time: date Thu Jun 25 18:09:37 2020-date Thu Jun 25 18:08:50 2020 s - exit: 256
cmsRun /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc820/cms/cmssw-patch/CMSSW_11_2_X_2020-06-25-1100/src/HLTrigger/Configuration/test/OnLine_HLT_Fake2.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Thu Jun 25 18:09:37 2020-date Thu Jun 25 18:08:50 2020 s - exit: 21504
cmsDriver.py RelVal -s HLT:Fake2,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_Fake2 --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2016 --processName=HLTRECO --filein file:RelVal_Raw_Fake2_MC.root --fileout file:RelVal_Raw_Fake2_MC_HLT_RECO.root : FAILED - time: date Thu Jun 25 18:09:37 2020-date Thu Jun 25 18:08:50 2020 s - exit: 21504
cmsDriver.py TTbar_Tauola_13TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW --mc --scenario=pp -n 10 --conditions auto:run3_mc_PIon --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run3 --fileout file:RelVal_Raw_PIon_MC.root : FAILED - time: date Thu Jun 25 18:09:56 2020-date Thu Jun 25 18:09:06 2020 s - exit: 256
cmsRun /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc820/cms/cmssw-patch/CMSSW_11_2_X_2020-06-25-1100/src/HLTrigger/Configuration/test/OnLine_HLT_PIon.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Thu Jun 25 18:09:56 2020-date Thu Jun 25 18:09:06 2020 s - exit: 21504
cmsDriver.py RelVal -s HLT:PIon,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run3_mc_PIon --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run3 --processName=HLTRECO --filein file:RelVal_Raw_PIon_MC.root --fileout file:RelVal_Raw_PIon_MC_HLT_RECO.root : FAILED - time: date Thu Jun 25 18:09:56 2020-date Thu Jun 25 18:09:06 2020 s - exit: 21504

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

rekovic committed:
SimL1Emulator sequence that was dropped.

cmsbuild commented:
The code-checks are being triggered in jenkins.

rekovic commented:
please test

rekovic committed:
SimL1Emulator_cff master default.

cmsbuild commented:
The code-checks are being triggered in jenkins.

rekovic commented:
please test

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-30325/16495

- This PR adds an extra 28KB to repository

cmsbuild commented:
New categories assigned: l1

@benkrikler,@rekovic you have been requested to review this Pull request/Issue and eventually sign? Thanks

cmsbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) cms-data/L1Trigger-L1TCalorimeter#22

cmsbuild commented:
Pull request #30325 was updated. @cmsbuild, @rekovic, @benkrikler, @kpedro88 can you please check and sign again.

cmsbuild commented:
+1
Tested at: c8605bdd677b4cdcc13adabffb7e9b8cb0f41fbc
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c806ed/7408/summary.html
CMSSW: CMSSW_11_2_X_2020-06-25-1100
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) cms-sw/cmssw#30363,cms-data/L1Trigger-L1TCalorimeter#22,cms-data/L1Trigger-L1TMuon#18,cms-sw/cmssw#30362

cmsbuild commented:
-1

Tested at: c8605bdd677b4cdcc13adabffb7e9b8cb0f41fbc

CMSSW: CMSSW_11_2_X_2020-06-25-1100
SCRAM_ARCH: slc7_amd64_gcc820
You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-e8f9f8/7418/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

When I ran the RelVals I found an error in the following workflows:
20034.0 step2
<pre>runTheMatrix-results/20034.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D35_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D35+RecoFullGlobal_2026D35+HARVESTFullGlobal_2026D35/step2_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D35_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D35+RecoFullGlobal_2026D35+HARVESTFullGlobal_2026D35.log

</pre>20434.0 step2
<pre>runTheMatrix-results/20434.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D41_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D41+RecoFullGlobal_2026D41+HARVESTFullGlobal_2026D41/step2_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D41_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D41+RecoFullGlobal_2026D41+HARVESTFullGlobal_2026D41.log

</pre>23234.0 step2
<pre>runTheMatrix-results/23234.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D49_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D49+RecoFullGlobal_2026D49+HARVESTFullGlobal_2026D49/step2_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D49_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D49+RecoFullGlobal_2026D49+HARVESTFullGlobal_2026D49.log

</pre>21234.0 step2
<pre>runTheMatrix-results/21234.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D44_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D44+RecoFullGlobal_2026D44+HARVESTFullGlobal_2026D44/step2_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D44_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D44+RecoFullGlobal_2026D44+HARVESTFullGlobal_2026D44.log

</pre>

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c806ed/7408/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 8 differences found in the comparisons
 - DQMHistoTests: Total files compared: 36
 - DQMHistoTests: Total histograms compared: 2778915
 - DQMHistoTests: Total failures: 7
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2778858
 - DQMHistoTests: Total skipped: 50
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 35 files compared)
 - Checked 152 log files, 16 edm output root files, 36 DQM output files

rekovic commented:
An example of failing job log messages indicate problems that have nothing to do with this PR [here](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-e8f9f8/7418/runTheMatrix-results/20034.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D35_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D35+RecoFullGlobal_2026D35+HARVESTFullGlobal_2026D35/step2_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D35_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D35+RecoFullGlobal_2026D35+HARVESTFullGlobal_2026D35.log), but instead with a TrackTrigger PR #30362

I am guessing there is a cross PR dependency in the testing introduced when testing the L1TTrackMatch PR
#30363, which uses both of these 2 PRs.

Try to test alone this PR.

rekovic commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) cms-data/L1Trigger-L1TCalorimeter#22

rekovic commented:
this looks like a bug in the bot.
Both this PR and PR #30362 (https://github.com/cms-sw/cmssw/pull/30362#issuecomment-649862609)
pointed to the same location of test results 
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-e8f9f8/7418/summary.html
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-e8f9f8/7418/summary.html
atttn @smuzaffar 

Testing alone this PR should pass, I expect.

smuzaffar commented:
@rekovic , please note that you had started a test to test multiple PRs together ( https://github.com/cms-sw/cmssw/pull/30363#issuecomment-649537943 ) . So when ever a test for PR #30363 will run, it will run for all of these cms-data/L1Trigger-L1TCalorimeter#22,cms-data/L1Trigger-L1TMuon#18,#30325,#30362 and results will points to same location.

cmsbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) cms-sw/cmssw#30363,cms-data/L1Trigger-L1TCalorimeter#22,cms-data/L1Trigger-L1TMuon#18,cms-sw/cmssw#30362

cmsbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) cms-sw/cmssw#30363,cms-data/L1Trigger-L1TCalorimeter#22,cms-data/L1Trigger-L1TMuon#18,cms-sw/cmssw#30362,cms-sw/cmssw#30245

silviodonato commented:
abort

cmsbuild commented:
Jenkins tests are aborted.

silviodonato commented:
test parameters
 * pull_request = cms-data/L1Trigger-L1TCalorimeter#22

silviodonato commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) cms-data/L1Trigger-L1TCalorimeter#22

silviodonato commented:
@rekovic I'm sorry I've made a mistake. I was confused from the two tests running at the same time

rekovic commented:
@silviodonato 
ok.np.

cmsbuild commented:
+1
Tested at: c8605bdd677b4cdcc13adabffb7e9b8cb0f41fbc
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c806ed/7422/summary.html
CMSSW: CMSSW_11_2_X_2020-06-25-2300
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

rekovic commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) cms-sw/cmssw#30363,cms-data/L1Trigger-L1TCalorimeter#22,cms-data/L1Trigger-L1TMuon#18,cms-sw/cmssw#30245

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c806ed/7422/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 36
 - DQMHistoTests: Total histograms compared: 2778915
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2778864
 - DQMHistoTests: Total skipped: 50
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 35 files compared)
 - Checked 152 log files, 16 edm output root files, 36 DQM output files

cmsbuild commented:
-1

Tested at: c8605bdd677b4cdcc13adabffb7e9b8cb0f41fbc

CMSSW: CMSSW_11_2_X_2020-06-25-2300
SCRAM_ARCH: slc7_amd64_gcc820
You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-a96035/7428/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

When I ran the RelVals I found an error in the following workflows:
23234.0 step2
<pre>runTheMatrix-results/23234.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D49_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D49+RecoFullGlobal_2026D49+HARVESTFullGlobal_2026D49/step2_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D49_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D49+RecoFullGlobal_2026D49+HARVESTFullGlobal_2026D49.log

</pre>

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

silviodonato commented:
@kpedro88 the test of #30325 only was ok https://github.com/cms-sw/cmssw/pull/30325#issuecomment-650095080

kpedro88 commented on L1Trigger/L1CaloTrigger/plugins/L1EGammaCrystalsEmulatorProducer.cc:
```diff
@@ -0,0 +1,1199 @@
+      throw cms::Exception("L1EGCrystalClusterEmulatorProducer");
+      continue;
+    }
+    std::vector<HcalDetId> hcId = theTrigTowerGeometry.detIds(hit.id());
```
repeated comment:
`const std::vector<HcalDetId>& hcId = theTrigTowerGeometry.detIds(hit.id());`

rekovic replied:
done

kpedro88 commented on L1Trigger/L1CaloTrigger/plugins/L1EGammaCrystalsEmulatorProducer.cc:
```diff
@@ -0,0 +1,1199 @@
+  float ECAL_tower_L2Card[n_links_GCTcard][n_towers_cardEta][3];  // 3 L2 cards send each 48 links with 17 towers
+  int iEta_tower_L2Card[n_links_GCTcard][n_towers_cardEta][3];    // 3 L2 cards send each 48 links with 17 towers
+  int iPhi_tower_L2Card[n_links_GCTcard][n_towers_cardEta][3];    // 3 L2 cards send each 48 links with 17 towers
+  float energy_cluster_L2Card[n_links_GCTcard][2][3];             // 3 L2 cards send each 48 links with 2 clusters
```
repeated comments:
magic numbers 2,3 should be named constants (throughout this section of code)
also, eliminate duplicate comments (just include each comment once)

rekovic replied:
done

kpedro88 commented on L1Trigger/L1CaloTrigger/python/test_L1EGCrystalClusterEmulator_cfg.py:
```diff
@@ -0,0 +1,76 @@
+
+from Configuration.StandardSequences.Eras import eras
+
+process = cms.Process("L1AlgoTest",eras.Phase2_trigger)
```
repeated comment:

please use the actual era for the phase 2 workflow being tested
(phase2_trigger is an internal use modifier, Phase2_trigger with that capitalization is not anything)

"please use the actual era for the phase 2 workflow being tested" does not mean "use the Phase2 era"

different scenarios have different associated eras, seen in https://github.com/cms-sw/cmssw/blob/f18f5a71e7278be55d457d17e38379b11cd04c19/Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py#L740

rekovic replied:
done.

kpedro88 commented on L1Trigger/L1CaloTrigger/interface/ParametricCalibration.h:
```diff
@@ -0,0 +1,46 @@
+    ParametricCalibration(const edm::ParameterSet& cpset);
+    static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+    float operator()(const float pt, const float abseta) const {
```
multiline implementations should typically be put in the .cc file
(unless e.g. it is desired to be inlined)

rekovic replied:
done

kpedro88 commented:
in addition to the few still unaddressed comments I've highlighted here, the code remains littered with magic numbers. But apparently everyone else is satisfied with allowing possible inconsistencies that could invalidate the results from O(100K) CPU hours spent for a full production. Oh well.

rekovic committed:
Magic numbers 2,3 and renaming of consts, Use correct era for expert test, Move method implementation to .cc.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-30325/16557

- This PR adds an extra 40KB to repository

cmsbuild commented:
Pull request #30325 was updated. @cmsbuild, @rekovic, @benkrikler, @kpedro88 can you please check and sign again.

rekovic committed:
Capture by const &.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-30325/16558

- This PR adds an extra 44KB to repository

cmsbuild commented:
Pull request #30325 was updated. @cmsbuild, @rekovic, @benkrikler, @kpedro88 can you please check and sign again.

rekovic commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) cms-data/L1Trigger-L1TCalorimeter#22

cmsbuild commented:
+1
Tested at: 1c3c7f6834b1c2a862d35e4b2e0297c56a59dd97
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c806ed/7466/summary.html
CMSSW: CMSSW_11_2_X_2020-06-28-0000
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c806ed/7466/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 36
 - DQMHistoTests: Total histograms compared: 2778915
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2778864
 - DQMHistoTests: Total skipped: 50
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 35 files compared)
 - Checked 152 log files, 16 edm output root files, 36 DQM output files

rekovic commented:
+1

kpedro88 commented:
+upgrade
(needs to be added to cms-bot package categories)

silviodonato commented:
Thank you for the reminder. @smuzaffar  @mrodozov could you assign it to l1 and upgrade?

silviodonato commented:
cms-bot package: https://github.com/cms-sw/cms-bot/pull/1336
cms-dist: https://github.com/cms-sw/cmsdist/pull/6004
(https://github.com/cms-data/L1Trigger-L1TCalorimeter/pull/22)

silviodonato commented:
merge

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will be automatically merged.

cmsbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) cms-sw/cmssw#30363,cms-data/L1Trigger-L1TCalorimeter#22,cms-data/L1Trigger-L1TMuon#18,cms-sw/cmssw#30245,cms-sw/cmssw#30458

cmsbuild commented:
-1
@rekovic This pull request cannot be automatically  merged, could you please rebase it?

You can see the log for git cms-merge-topic here: 
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-30325/7502/git-merge-result

cmsbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) cms-sw/cmssw#30363,cms-data/L1Trigger-L1TCalorimeter#22,cms-data/L1Trigger-L1TMuon#18,cms-sw/cmssw#30245,cms-sw/cmssw#30458

cmsbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) cms-sw/cmssw#30363,cms-data/L1Trigger-L1TCalorimeter#22,cms-data/L1Trigger-L1TMuon#18,cms-sw/cmssw#30245,cms-sw/cmssw#30458

cmsbuild commented:
+1
Tested at: 1c3c7f6834b1c2a862d35e4b2e0297c56a59dd97
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-4a912c/7514/summary.html
CMSSW: CMSSW_11_2_X_2020-06-29-2300
SCRAM_ARCH: slc7_amd64_gcc820

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - #30451 @JamminJones: added esConsumes to modules in RecoLocalMuon/GEMSegment
 - #30450 @JamminJones: added esConsumes to modules in RecoLocalMuon/GEMRecHit
 - #30458 @cms-sw: Revert "Run4-hgx247 Add some new features for HGCal"

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-4a912c/7514/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-4a912c/7514/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-4a912c/7514/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 36
 - DQMHistoTests: Total histograms compared: 2778915
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2778864
 - DQMHistoTests: Total skipped: 50
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 35 files compared)
 - Checked 152 log files, 16 edm output root files, 36 DQM output files

cmsbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) cms-sw/cmssw#30363,cms-data/L1Trigger-L1TCalorimeter#22,cms-data/L1Trigger-L1TMuon#18,cms-sw/cmssw#30245,cms-sw/cmssw#30458

cmsbuild commented:
+1
Tested at: 1c3c7f6834b1c2a862d35e4b2e0297c56a59dd97
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-4a912c/7551/summary.html
CMSSW: CMSSW_11_2_X_2020-06-30-2300
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-4a912c/7551/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 4 differences found in the comparisons
 - DQMHistoTests: Total files compared: 37
 - DQMHistoTests: Total histograms compared: 2784120
 - DQMHistoTests: Total failures: 5
 - DQMHistoTests: Total nulls: 1
 - DQMHistoTests: Total successes: 2784064
 - DQMHistoTests: Total skipped: 50
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: -0.004 KiB( 36 files compared)
 - DQMHistoSizes: changed ( 10224.0 ): -0.004 KiB       MessageLogger/Warnings
 - Checked 154 log files, 17 edm output root files, 37 DQM output files
